File: 563/1.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <map>
#include <set>
#include <unordered_map>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int countLeft = 0, countRight = 0;
    int getVal(TreeNode* node, int count){
        if(!node) return 0;
        count += node->val;
        return getVal(node->left, count)+getVal(node->right, count);
    }
    int findTilt(TreeNode* root) {
        return abs(getVal(root->right) - getVal(root->left));
    }
};

int main(){

    int n, m, num;

    Solution solution;
    int res = solution.numSpecial(mat);

    cout << res << endl;

    return 0;
}

File: 226/1.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def traverse(self, node: Optional[TreeNode]):
        node.left, node.right = node.right, node.left
        if(node.left): 
            self.traverse(node.left)
        if(node.right):
            self.traverse(node.right)

    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.traverse(root)
        return root